lib/theme_check/shopify_liquid/source_index/base_entry.rb

```diff
@@ -7,19 +7,19 @@ class BaseEntry
         attr_reader :hash
 
         def initialize(hash = {})
-          @hash = hash
+          @hash = hash || {}
         end
 
         def name
           hash['name']
         end
 
         def summary
-          hash['summary']
+          hash['summary'] || ''
         end
 
         def description
-          hash['description']
+          hash['description'] || ''
```

Comment:
Not sure in which PR you want to tackle this (did not find a linked TODO), so I'll leave it here :-)
The descriptions are still kinda "rough"/unformatted: they contain symbols like …
- `--`
- `> `
- `<br><br>`

![14-50-vg84t-eh4ao](https://user-images.githubusercontent.com/12963280/201611818-54ef109f-41b2-4c76-ae85-fe84b0e13109.jpg)

Reply:
Also maybe related to this:
When clicking on links…
![14-42-xpxs4-m538o](https://user-images.githubusercontent.com/12963280/201614866-c45ea0a1-a056-45d0-8000-2b6265b076bf.jpg)
… VS Code opens an invalid URL:
![14-40-dbu2m-lngn4](https://user-images.githubusercontent.com/12963280/201614966-5155b66c-58b7-4bf2-807f-fdc2df5daa67.jpg)